resources/type-system/subtyping.md

```diff
@@ -247,10 +283,71 @@ Right union introduction:
  - `S <: T0 | T1` if `S <: T0` or `S <: T1`
 
 Left intersection introduction:
- - `X & S <: T` if `X <: T` or `S <: T`
+ - `S0 & S1 <: T` if `S0 <: T` or `S1 <: T`
 
 Right intersection introduction:
- - `S <: X & T` if `S <: X` and `S <: T`
+ - `S <: T0 & T1` if `S <: T0` and `S <: T1`
+
+Axioms for `Null` and `Object`:
+
+Left non-nullable `Null`:
+ - `Null! <: T`
```

Comment:
This would allow us to prove `Never <: Null!` and `Null! <: Never`, but not `Null! == Never`. I guess that's because you think it would actually be more trouble and not more useful to introduce some sort of normalization?

Reply:
I don't know what you mean by `Null! == Never` other than that they are mutual subtypes, which this allows us to prove.  I'm open to trying to specify a normalization process that reduces mutual subtyping to normalization + syntactic equality, but such a normalization process, if it exists, is an implementation strategy.  Unless you're proposing to define an equality that is more restrictive than mutual subtyping?  Note that if not, then your normalization strategy needs to deal with things like `X extends Never` `Y extends Never`, which are mutual subtypes.  Probably that means that you have to normalize them to `Never`?  In any case, lots of bits to work out.